presto-hive/src/test/java/com/facebook/presto/hive/TestPrestoS3FileSystem.java

```diff
@@ -169,6 +176,62 @@ public void testReadForbidden()
     }
 
     @Test
+    public void testCreateWithNonexistentStagingDirectory()
+        throws Exception
+    {
+        java.nio.file.Path tmpdir = Paths.get(StandardSystemProperty.JAVA_IO_TMPDIR.value());
+        java.nio.file.Path stagingParent = Files.createTempDirectory(tmpdir, "staging");
```

Comment:
Yeah, "staging" was probably a weird prefix to choose here, when this is in fact the parent of the actual staging directory. Good suggestion calling it just "test".